python/pip_install/tools/wheel_installer/wheel.py

```diff
@@ -94,13 +129,27 @@ def from_tag(cls, tag: str) -> "Platform":
         )
 
     @classmethod
-    def from_string(cls, platform: str) -> "Platform":
-        os, _, arch = platform.partition("_")
+    def from_string(cls, platform: Union[str, List[str]]) -> List["Platform"]:
```

Comment:
Small simplification: normalize the value to a list, then operate with the list value uniformly. Since we have to potentially loop over the values anyways to match the `*` cases, we can just always loop:

```
values = [value] if isinstance(str, value) else value

for value in values:
  if value == "host": result.update(...)
  elif value == "all": result.update(...)
  elif <star case> : ...
  return result
``